### PR TITLE
Added possibility to shuffle lists

### PR DESCRIPTION
**Pull request by votruk**

### CODE REVIEW

**Comment by passsy** [comment]
Use the same signature as darts `shuffle`

```dart
void shuffle([Random random]);
```

Using a pre-seeded Random allows you to add tests. Shuffle with two different seeds and compare that the result isn't the same.

**Comment by votruk** [comment]
What do you think, is it enough to test just for a case with different seeds?

File: test/collection/list_mutable_extensions_test.dart [hunk 1]
```diff
@@ -153,4 +156,16 @@ void testList(
...
   });
+
+  group("shuffle", () {
+    test("shuffle shuffles items in a list with provided Random object", () {
+      final firstList = mutableListOf(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+      firstList.shuffle(Random(1));
+      final secondList = mutableListOf(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+      secondList.shuffle(Random(2));
+      expect(
+          const ListEquality().equals(firstList.asList(), secondList.asList()),
```
Comment (passsy) [inline test/collection/list_mutable_extensions_test.dart]: You can compare `firstList` and `secondList` directly without the ListEquality helper. That's one of the core features of ktdart collections :wink:

```dart
expect(firstList, isNot(equals(secondList)));
```

(Don't forget to remove the `collection` dependency)

Comment (votruk) [reply test/collection/list_mutable_extensions_test.dart]: Great to know!

**Pull request merged**